src/CoreWCF.MSMQ/src/CoreWCF/Channels/MsmqNetcoreTransport.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreWCF.Configuration;
+using CoreWCF.Queue.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using MSMQ.Messaging;
+using MSMQM = MSMQ.Messaging;
+
+namespace CoreWCF.Channels
+{
+    public class MsmqNetcoreTransport : IQueueTransport, IDisposable
+    {
+        private readonly Uri _baseAddress;
+        private readonly MessageQueue _messageQueue;
+        private readonly TimeSpan _queueReceiveTimeOut;
+        private readonly IDeadLetterQueueMsmqSender _deadLetterQueueSender;
+        private readonly ILogger<MsmqNetcoreTransport> _logger;
+
+        public MsmqNetcoreTransport(QueueOptions options, IServiceDispatcher serviceDispatcher, IServiceProvider serviceProvider)
+        {
+            _deadLetterQueueSender = serviceProvider.GetRequiredService<IDeadLetterQueueMsmqSender>();
+            _baseAddress = serviceDispatcher.BaseAddress;
+            string nativeQueueName = MsmqQueueNameConverter.GetMsmqFormatQueueName(options.QueueName);
+            _messageQueue = new MessageQueue(nativeQueueName);
```

Comment:
The queue name shouldn't come from QueueOptions, it should be derived from the URL (which is in BaseAddress). There's only a single QueueOptions as well, which means you could only pull from a single queue which might not be the case.